From b059a5d8824f2a14160f7e0f879936f99f5c3aa9 Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Fri, 28 Jul 2023 09:59:14 -0300
Subject: [PATCH] UPSTREAM: <carry>: Add OpenShift tooling, images, configs and
 docs

Update the list of tests that should be skipped.
---
 openshift-hack/e2e/annotate/rules.go | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/openshift-hack/e2e/annotate/rules.go b/openshift-hack/e2e/annotate/rules.go
index ce2e8c14913..fc537f38f1d 100644
--- a/openshift-hack/e2e/annotate/rules.go
+++ b/openshift-hack/e2e/annotate/rules.go
@@ -19,6 +19,9 @@ var (
 			`\[Feature:InPlacePodVerticalScaling\]`,
 			`\[Feature:RecoverVolumeExpansionFailure\]`,
 			`\[Feature:SELinux\]`,
+			`\[Feature:SidecarContainers\]`,
+			`\[Feature: PersistentVolumeLastPhaseTransitionTime\]`,
+			`\[Feature:WatchList\]`,
 		},
 		// tests for features that are not implemented in openshift
 		"[Disabled:Unimplemented]": {
@@ -153,9 +156,19 @@ var (
 			`\[sig-network\] Connectivity Pod Lifecycle should be able to connect to other Pod from a terminating Pod`, // TODO(network): simple test in k8s 1.27, needs investigation
 			`\[sig-cli\] Kubectl client Kubectl prune with applyset should apply and prune objects`,                    // TODO(workloads): alpha feature in k8s 1.27. It's failing with `error: unknown flag: --applyset`. Needs investigation
 
-			// https://issues.redhat.com/browse/OCPBUGS-13392
-			`\[sig-network\] NetworkPolicyLegacy \[LinuxOnly\] NetworkPolicy between server and client should enforce policy to allow traffic only from a pod in a different namespace based on PodSelector and NamespaceSelector`,
-			`\[sig-network\] NetworkPolicyLegacy \[LinuxOnly\] NetworkPolicy between server and client should enforce updated policy`,
+			// https://issues.redhat.com/browse/OCPBUGS-16760
+			`\[Feature:NodeLogQuery\]`,
+
+			// https://issues.redhat.com/browse/OCPBUGS-16922
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should reject mutating webhook configurations with invalid match conditions`,
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should be able to deny pod and configmap creation`,
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should be able to create and update validating webhook configurations with match conditions`,
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should be able to create and update mutating webhook configurations with match conditions`,
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should reject validating webhook configurations with invalid match conditions`,
+			`AdmissionWebhook \[Privileged:ClusterAdmin\] should mutate everything except 'skip-me' configmaps`,
+
+			// https://issues.redhat.com/browse/OCPBUGS-17194
+			`\[sig-node\] ImageCredentialProvider \[Feature:KubeletCredentialProviders\] should be able to create pod with image credentials fetched from external credential provider`,
 		},
 		// tests that may work, but we don't support them
 		"[Disabled:Unsupported]": {
-- 
2.41.0

